pkg/clusterversion/README.md

```diff
@@ -0,0 +1,249 @@
+
+## Master branch changes
+
+The changes below happen on the `master` branch after forking the release, and
+are not backported to the release branch.
+
+### M.1: Bump min supported
+
+**What**: This change advances the `MinSupported` version. This is normally two
+versions behind the current release (we support "skipping" a version during
+upgrade as an experimental feature). Once `MinSupported` is advanced, all older
+non-permanent version keys are no longer necessary - Cockroach code will never
+deal with a cluster below `MinSupported`.
+
+**When**: Any time after forking the release branch.
+
+**Checklist**:
+
+- [ ] Advance `MinVersion` to the previous release (e.g. `V23.2`)
+
+- [ ] Rename all non-permanent version keys below `MinVersion`, prepending
+  `TODO_Delete_` to them.
+  <details><summary>Additional context</summary>
+  The way we name version gates can potentially be misleading:
+
+   - Gates beginning with `{MAJOR_SERIES}*` help migrate the system from
+     `{MAJOR_SERIES}-1` to `{MAJOR_SERIES}`. These gates have cluster versions
+     **below** that of the final release version for `{MAJOR_SERIES}`: they use the
+     previous series major and minor and a non-zero (and even) internal value. For
+     example, `V23_2SomeFeature` would be associated with a version like `v23.1-x`
+     (e.g. `v23.1-8`).
+
+   - Once `MinSupported` is advanced to `{MAJOR_SERIES}`  we no longer need these
+     `{MAJOR_SERIES}*` gates. For example, a gate like `V23_2SomeFeature` is used
+     during 23.1 → 23.2 upgrade; once `MinSupported = V23_2` this gate will always be
+     "active" in any clusters our code has to deal with.
+  </details>
+
+- [ ] Remove logictest config that bootstraps to the old minimum supported
+  version (and run `./dev gen testlogic`).
+
+- [ ] File issue(s) to remove `TODO_Delete_` uses and simplify related code; assign
+  to relevant teams (depending on which top-level packages use such gates).
+  <details><summary>For the overachiever</summary>
+  Historically, these cleanup issues have not received much attention and the code
+  was carried over for a long time. Feel free to ping individuals or do some of
+  the cleanup directly (in separate PRs). The cleanup comes down to simplifying the
+  code based on the knowledge that `IsActive()` will always return `true` for
+  obsolete gates. If simplifying the code becomes non-trivial, error on the side
+  of just removing `IsActive()` calls and leaving TODOs for further cleanup.
+  </details>
+  
+- [ ] Regenerate expected test data results as needed; file issues for any
+  skipped tests. Note that upgrade tests in `pkg/upgrade/upgrades` use
+  `clusterversion.SkipWhenMinSupportedVersionIsAtLeast()` so they can be removed
+  later (as part of cleaning up the obsolete gates).
+
+**Example PR:** [#112122](https://github.com/cockroachdb/cockroach/pull/112122)
+(note that this particular PR required a lot of fixes that will no longer be
```

Comment:
We can probably update this example with the next one. I bet it will be much simpler.